vtkext/private/module/vtkF3DTAAResolvePass.cxx

```diff
@@ -0,0 +1,191 @@
+    this->HistoryTexture->SetMagnificationFilter(vtkTextureObject::Linear);
+    this->HistoryTexture->SetWrapS(vtkTextureObject::ClampToEdge);
+    this->HistoryTexture->SetWrapT(vtkTextureObject::ClampToEdge);
+    renderer->GetTiledSizeAndOrigin(&size[0], &size[1], &pos[0], &pos[1]);
+    this->HistoryTexture->Allocate2D(size[0], size[1], 4, VTK_FLOAT);
+    this->HistoryInitialized = false;
+  }
+  this->HistoryTexture->Resize(size[0], size[1]);
+
+  if (this->ColorTexture == nullptr)
+  {
+    this->ColorTexture = vtkSmartPointer<vtkTextureObject>::New();
+    this->ColorTexture->SetContext(renWin);
+    this->ColorTexture->SetMinificationFilter(vtkTextureObject::Linear);
+    this->ColorTexture->SetMagnificationFilter(vtkTextureObject::Linear);
+    this->ColorTexture->SetWrapS(vtkTextureObject::ClampToEdge);
+    this->ColorTexture->SetWrapT(vtkTextureObject::ClampToEdge);
+    this->ColorTexture->Allocate2D(size[0], size[1], 4, VTK_FLOAT);
+  }
+  this->ColorTexture->Resize(size[0], size[1]);
+
+  if (this->FrameBufferObject == nullptr)
+  {
+    this->FrameBufferObject = vtkSmartPointer<vtkOpenGLFramebufferObject>::New();
+    this->FrameBufferObject->SetContext(renWin);
+  }
+
+  renWin->GetState()->PushFramebufferBindings();
+  this->RenderDelegate(
+    s, size[0], size[1], size[0], size[1], this->FrameBufferObject, this->ColorTexture);
+  renWin->GetState()->PopFramebufferBindings();
+
+  if (this->QuadHelper && this->QuadHelper->ShaderChangeValue < this->GetMTime())
+  {
+    this->QuadHelper = nullptr;
+  }
+
+  if (!this->QuadHelper)
+  {
+    std::string TAAResolveFS = vtkOpenGLRenderUtilities::GetFullScreenQuadFragmentShaderTemplate();
+    vtkShaderProgram::Substitute(TAAResolveFS, "//VTK::FSQ::Decl",
+      "uniform sampler2D colorTexture;\n"
+      "uniform sampler2D historyTexture;\n"
+      "uniform float blendFactor;\n"
+      "//VTK::FSQ::Decl");
+
+    vtkShaderProgram::Substitute(TAAResolveFS, "//VTK::FSQ::Impl",
+      "vec4 current = texture(colorTexture, texCoord);\n"
+      "vec4 history = texture(historyTexture, texCoord);\n"
+      "gl_FragData[0] = mix(current, history, blendFactor);\n"
+      "//VTK::FSQ::Impl");
+    this->QuadHelper =
+      std::make_shared<vtkOpenGLQuadHelper>(renWin, nullptr, TAAResolveFS.c_str(), nullptr);
+    this->QuadHelper->ShaderChangeValue = this->GetMTime();
+  }
+  else
+  {
+    renWin->GetShaderCache()->ReadyShaderProgram(this->QuadHelper->Program);
+  }
+
+  if (!this->QuadHelper->Program || !this->QuadHelper->Program->GetCompiled())
+  {
+    vtkErrorMacro("Couldn't build the shader program.");
+    return;
+  }
+
+  this->HistoryTexture->Activate();
+  this->ColorTexture->Activate();
+  this->QuadHelper->Program->SetUniformi("colorTexture", this->ColorTexture->GetTextureUnit());
+  this->QuadHelper->Program->SetUniformi("historyTexture", this->HistoryTexture->GetTextureUnit());
+
+  float blendFactor = this->HistoryInitialized ? 0.9f : 0.0f;
```

Comment:
should that 0.9 be a const expr ?

Reply:
Maybe in the future it could be opened to the user, but I agree it should be constexpr at the moment

Reply:
`blendFactor` depends on `HistoryInitialized`, which is not determined by compile time. In this case it can't be declared as `constexpr`, I've updated to `const`